### ISSUE TITLE
Update adorn_ functions documentation to show how to use commas with `...` specification

### ISSUE THREAD

**Issue by francisbarton** [sfirke/janitor#384]
## Bug report
`adorn_totals()` should allow the user to select (by inclusion or exclusion) the variables for which to attach totals.

---

The [`adorn_totals` man page](https://sfirke.github.io/janitor/reference/adorn_totals.html) states that `...` allows the user to "manually specify which columns should be included" using "a [tidyselect](https://tidyselect.r-lib.org/reference/language.html) specification".

But in my reprex below, where I attempt to specify, using tidyselect grammar, particular columns *not* to total, I get an error.

If there is a different way of specifying certain columns not to include in `adorn_totals()` then please let me know.

``` r
library(dplyr, warn.conflicts = FALSE)
library(tibble)
library(janitor, warn.conflicts = FALSE)

as.data.frame(datasets::state.x77) %>% 
  tibble::rownames_to_column("state") %>% 
  janitor::clean_names() %>% 
  dplyr::filter(population < 500) %>% 
  dplyr::select(-c(murder, frost)) %>% 
  janitor::adorn_totals()
#>    state population income illiteracy life_exp hs_grad   area
#>   Alaska        365   6315        1.5    69.31    66.7 566432
#>  Vermont        472   3907        0.6    71.64    57.1   9267
#>  Wyoming        376   4566        0.6    70.29    62.9  97203
#>    Total       1213  14788        2.7   211.24   186.7 672902

as.data.frame(datasets::state.x77) %>% 
  tibble::rownames_to_column("state") %>% 
  janitor::clean_names() %>% 
  dplyr::filter(population < 500) %>% 
  dplyr::select(-c(murder, frost)) %>% 
  janitor::adorn_totals(where = "row", -c(illiteracy, life_exp))
#> Error in .f(.x[[i]], ...): object 'illiteracy' not found

as.data.frame(datasets::state.x77) %>% 
  tibble::rownames_to_column("state") %>% 
  janitor::clean_names() %>% 
  dplyr::filter(population < 500) %>% 
  dplyr::select(-c(murder, frost)) %>% 
  janitor::adorn_totals(!matches("illiteracy"))
#> Error: `matches()` must be used within a *selecting* function.
#> i See <https://tidyselect.r-lib.org/reference/faq-selection-context.html>.
```

<sup>Created on 2020-07-03 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by sfirke** [comment]
Unfortunately, to specify values for `...` you have to specify all of the other argument values.  You can't say, "... = !matches("illiteracy")`.  See [this StackOverflow Q ](https://stackoverflow.com/questions/60760245/specify-the-dots-argument-when-calling-a-tidyselect-using-function-without-needi) for more background.

You have to do this, with a bunch of commas first to address the preceding arguments:
```
as.data.frame(datasets::state.x77) %>% 
  tibble::rownames_to_column("state") %>% 
  janitor::clean_names() %>% 
  dplyr::filter(population < 500) %>% 
  dplyr::select(-c(murder, frost)) %>% 
  janitor::adorn_totals(,,,,!matches("illiteracy"))

   state population income illiteracy life_exp hs_grad   area
  Alaska        365   6315        1.5    69.31    66.7 566432
 Vermont        472   3907        0.6    71.64    57.1   9267
 Wyoming        376   4566        0.6    70.29    62.9  97203
   Total       1213  14788          -   211.24   186.7 672902
```

I will leave this issue open though as I think the documentation could be improved.  I show the `,,,,` in the [tabyls vignette](http://sfirke.github.io/janitor/articles/tabyls.html), but it deserves a mention in the documentation of every `adorn_` function that takes `...`.

**Comment by francisbarton** [comment]
D'oh! Yes of course, that makes sense. Thanks.

**Comment by francisbarton** [comment]
I can run the example in your reply above just fine, but I'm persistently getting an error/failure when knitting an Rmd with this syntax in the real world.

Here's the snippet of code that's causing the issue (sorry I haven't made a reprex):

```
df %>%
  [series of pipes] %>%
  arrange(desc(density)) %>%  # density is a numeric column and `arrange` succeeds here just fine
  janitor::adorn_totals(,,,,!matches("density")) # 4 commas as `dat` is implicit from the pipe
```

and here's the error that `knitr` throws:
```
Error: Assigned data `col_totals[1, ]` must be compatible with existing data.
i Error occurred for column `density`.
x Can't convert <character> to <double>.
Backtrace:
     x
  1. +-rmarkdown::render_site(...)
  2. | \-generator$render(...)
  3. |   \-base::sapply(...)
  4. |     \-base::lapply(X = X, FUN = FUN, ...)
  5. |       \-rmarkdown:::FUN(X[[i]], ...)
  6. |         \-rmarkdown:::render_one(...)
  7. |           +-base::suppressMessages(rmarkdown::render(...))
  8. |           | \-base::withCallingHandlers(...)
  9. |           \-rmarkdown::render(...)
 10. |             \-knitr::knit(knit_input, knit_output, envir = envir, quiet = quiet)
 11. |               \-knitr:::process_file(text, output)
 12. |                 +-base::withCallingHandlers(...)
 13. |                 +-knitr:::process_group(group)
 14. |                 \-knitr:::process_group.block(group)
 15. |                   \-knitr:::call_block(x)
 16. |                     \-knitr:::block_exec(params)
 17. |   

Execution halted
```

Does that error make any sense to you, Sam? I can't understand why it's trying to convert a character to a double.

**Comment by francisbarton** [comment]
If I save the following code as an R file or Rmd file

```
library(dplyr, warn.conflicts = FALSE)
library(tibble)
library(janitor, warn.conflicts = FALSE)

as.data.frame(datasets::state.x77) %>% 
  tibble::rownames_to_column("state") %>% 
  janitor::clean_names() %>% 
  dplyr::filter(population < 500) %>% 
  dplyr::select(-c(murder, frost)) %>% 
  janitor::adorn_totals(,,,,!matches("illiteracy"))
```
it runs fine, but the adorn_totals line is flagged by the code diagnostics within RStudio with the message "missing argument to function call".

**Comment by sfirke** [comment]
I think your first comment today, with `Error: Assigned data `col_totals[1, ]` must be compatible with existing data.`, is a separate issue and might truly be a bug.  There's a type mismatch happening internally when janitor creates the totals row to append.

A reprex would be helpful there, I tried for a minute to reproduce it myself and couldn't.  Could you take the data.frame that's producing the error, see if you can select just a few columns and slice a few rows and still have it produce that error - then scramble or rename anything sensitive and call `dput` on it, sharing the result here?  Or, if you want to send me something that you don't want the world to see, you can email me an object (my email address is an image at the bottom of the main repo page).

**Comment by francisbarton** [comment]
``` r
library(dplyr)
structure(list(wd19nm = c("Clifton", "Holgate", "Acomb"),
               popn_total = c(10068, 13298, 8726),
               density = c(6480, 4328, 3916)),
          row.names = c(NA, -3L),
          class = c("tbl_df", "tbl", "data.frame")) %>%
  janitor::adorn_totals(,,,,!matches("density"))
#> Error: Assigned data `col_totals[1, ]` must be compatible with existing data.
#> i Error occurred for column `density`.
#> x Can't convert <character> to <double>.
```

<sup>Created on 2020-07-10 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by sfirke** [comment]
Yep that's the issue, specifically when the input is a tibble.  I've made this a separate issue from the documentation fix, it's now #388 .
